jackson-modules/jackson-conversions-3/src/test/java/com/baeldung/jackson/absentvsnull/JacksonAbsentVsNullUnitTest.java

```diff
@@ -0,0 +1,130 @@
+package com.baeldung.jackson.absentvsnull;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+
+import com.baeldung.jackson.absentvsnull.model.Sample;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.exc.MismatchedInputException;
+
+class JacksonAbsentVsNullUnitTest {
+
+    private static final String ID_FIELD = "id";
+    private static final String PRIMITIVE_FIELD = "amount";
+    private static final ObjectMapper DEFAULT_MAPPER = new ObjectMapper();
+
+    private static final Sample ZERO_ARG_SAMPLE = new Sample();
+    private static final Sample FULL_SAMPLE = new Sample();
+
+    @AfterEach
+    void setup() {
+        List<String> keys = new ArrayList<>();
+        keys.add("foo");
+        keys.add("bar");
+
+        List<Integer> values = new ArrayList<>();
+        values.add(1);
+        values.add(2);
+
+        FULL_SAMPLE.setId(1l);
+        FULL_SAMPLE.setName(ID_FIELD);
+        FULL_SAMPLE.setAmount(3);
+        FULL_SAMPLE.setKeys(keys);
+        FULL_SAMPLE.setValues(values);
+    }
+
+    @Test
+    final void whenSerializingWithDefaults_thenNullValuesIncluded() {
+        Sample sample = ZERO_ARG_SAMPLE;
+        Map<String, Serializable> map = DEFAULT_MAPPER.convertValue(sample, SampleUtils.MAP_TYPE);
+
+        assertEquals(ZERO_ARG_SAMPLE.getAmount(), map.get("amount"));
+        assertTrue(map.containsKey(ID_FIELD));
+        assertNull(map.get(ID_FIELD));
+    }
+
+    @Test
+    final void whenDeserializingToMapWithDefaults_thenAbsentFieldsAreIgnored() throws JsonProcessingException {
+        String json = "{\"values\": [2], \"keys\": []}";
+        Map<String, Serializable> map = DEFAULT_MAPPER.readValue(json, SampleUtils.MAP_TYPE);
+
+        assertFalse(map.containsKey(PRIMITIVE_FIELD));
```

Comment:
It seems like there is some context missing with these assertions. For example, the `json` has `values` and `keys`, but the test here is about `amount`. This is a bit jarring for the reader because these don't feel related. An assertion like:

```java
Set<String> keySet = map.keySet();
assertTrue(keySet.size() == 2);
assertTrue(keySet.containsAll(List.of("keys", "values")));
```

might make more sense since it remains in the context of what is declared in the test givens.

Another might be:

```java
assertTrue(map.size() == 2);
assertTrue(map.containsKey("keys"));
assertTrue(map.containsKey("values"));
```